File: D/944-delete-columns-to-make-sorted/main.go
package main

import "fmt"

func minDeletionSize(A []string) int {
	S := len(A[0])
	c := 0
	for i := 0; i < S; i++ {
		b := A[0][i]
		isValid := true
		for j := 1; j < len(A); j++ {
			if A[j][i] < b {
				isValid = false
				break
			}
			b = A[j][i]
		}
		if !isValid {
			c++
		}
	}
	return c
}

func main() {
	fmt.Println("a")
}

File: D/944-delete-columns-to-make-sorted/main.rs
impl Solution {
  pub fn min_deletion_size(strs: Vec<String>) -> i32 {
    let mut ans: i32 = 0;
    let bbs = strs.iter().map(|x| x.as_bytes().to_vec()).collect::<Vec<Vec<u8>>>();
    for c in 0..bbs[0].len() {
      let mut valid: bool = true;
      for r in 1..bbs.len() {
        if bbs[r][c] < bbs[r - 1][c] {
          valid = false;
          break;
        }
      }
      if !valid {
        ans += 1;
      }
    }
    ans
  }
}
